host/client_integration_test.go

```diff
@@ -442,6 +452,119 @@ func (s *clientIntegrationSuite) TestClientDataConverter_WithChild() {
+	parentWorkflowId := "client-integration-too-many-child-workflows"
+	blockingChildWorkflow := func(ctx workflow.Context) error {
+		workflow.SignalExternalWorkflow(ctx, parentWorkflowId, "", "blocking-child-started", nil)
+		workflow.GetSignalChannel(ctx, "unblock-child").Receive(ctx, nil)
+		return nil
+	}
+	childWorkflow := func(ctx workflow.Context) error {
+		return nil
+	}
+
+	maxPendingChildWorkflows := s.maxPendingChildExecutions
+	parentWorkflow := func(ctx workflow.Context) error {
+		childStarted := workflow.GetSignalChannel(ctx, "blocking-child-started")
+		for i := 0; i < maxPendingChildWorkflows; i++ {
+			childOptions := workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+				WorkflowID: fmt.Sprintf("child-%d", i+1),
+			})
+			workflow.ExecuteChildWorkflow(childOptions, blockingChildWorkflow)
+		}
+		for i := 0; i < maxPendingChildWorkflows; i++ {
+			childStarted.Receive(ctx, nil)
+		}
+		return workflow.ExecuteChildWorkflow(workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+			WorkflowID: fmt.Sprintf("child-%d", maxPendingChildWorkflows+1),
+		}), childWorkflow).Get(ctx, nil)
+	}
+	s.worker.RegisterWorkflow(blockingChildWorkflow)
+	s.worker.RegisterWorkflow(childWorkflow)
+	s.worker.RegisterWorkflow(parentWorkflow)
+
+	timeout := time.Minute * 5
+	ctx, cancel := rpc.NewContextWithTimeoutAndVersionHeaders(timeout)
+	defer cancel()
+	options := sdkclient.StartWorkflowOptions{
+		ID:                 parentWorkflowId,
+		TaskQueue:          s.taskQueue,
+		WorkflowRunTimeout: timeout,
+	}
+
+	// start the parent workflow
+	future, err := s.sdkClient.ExecuteWorkflow(ctx, options, parentWorkflow)
+	s.NoError(err)
+
+	// wait until the workflow task to create the child workflow is retried
+	s.NoError(backoff.Retry(func() error {
+		time.Sleep(time.Second)
+		result, err := s.sdkClient.DescribeWorkflowExecution(ctx, parentWorkflowId, "")
+		s.NoError(err)
+		if result.PendingWorkflowTask != nil {
+			if result.PendingWorkflowTask.Attempt > 1 {
+				s.Len(result.PendingChildren, s.maxPendingChildExecutions)
+				return nil
+			}
+		}
+		return fmt.Errorf("the task to create the child workflow was never retried")
+	}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx)))
+
+	// unblock the last child, allowing it to complete, which lowers the number of pending child workflows
+	err = s.sdkClient.SignalWorkflow(
+		ctx,
+		fmt.Sprintf("child-%d", maxPendingChildWorkflows),
+		"",
+		"unblock-child",
+		nil,
+	)
+	s.NoError(err)
+
+	// verify that the parent workflow completes soon after the number of pending child workflows drops
+	err = backoff.Retry(func() error {
+		return future.Get(ctx, nil)
+	}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
+	s.NoError(err)
+
+	// verify that each still-blocked child eventually terminates now that the parent is closed
+	for i := 0; i < maxPendingChildWorkflows-1; i++ {
```

Comment:
I think this covers the parent child policy. Not sure if we need this in this case?